src/main/java/org/folio/circulation/resources/RequestFromRepresentationService.java

```diff
@@ -188,19 +191,45 @@ private CompletableFuture<Result<Request>> fetchItemAndLoanForPageTlrReplacement
   }
 
   private Result<Request> findItemForPageTlr(Request request) {
-    return getFirstAvailableItem(request)
+    return getFirstAvailableItemClosestToPickupServicePoint(request)
       .map(request::withItem)
       .map(Result::succeeded)
       .orElseGet(() -> failedValidation(
         "Cannot create page TLR for this instance ID - no available items found", INSTANCE_ID,
         request.getInstanceId()));
   }
 
-  private Optional<Item> getFirstAvailableItem(Request request) {
+  private Optional<Item> getFirstAvailableItemClosestToPickupServicePoint(Request request) {
+    List<Item> items = getAvailableItems(request).collect(toList());
+    String servicePointId = request.getPickupServicePointId();
+
+    Optional<Item> byLocation = searchItem(items, servicePointId, Location::getId);
+    Optional<Item> byLibrary = searchItem(items, servicePointId, Location::getLibraryId);
+    Optional<Item> byCampus = searchItem(items, servicePointId, Location::getCampusId);
+    Optional<Item> byInstitution = searchItem(items, servicePointId, Location::getInstitutionId);
+
+    return Stream.of(byLocation, byLibrary, byCampus, byInstitution)
+      .filter(Optional::isPresent)
+      .map(Optional::get)
+      .findFirst()
+      .or(() -> getAvailableItems(request).findFirst());
+  }
+
+  private Stream<Item> getAvailableItems(Request request) {
     return request.getInstanceItems().stream()
-        .filter(Objects::nonNull)
-        .filter(item -> ItemStatus.AVAILABLE == item.getStatus())
-        .findFirst();
+      .filter(Objects::nonNull)
+      .filter(item -> ItemStatus.AVAILABLE == item.getStatus());
+  }
+
+  private Optional<Item> searchItem(List<Item> items, String servicePointId,
+    Function<Location, String> locationFunction) {
+
+    return items.stream()
+      .filter(item -> {
+        Location location = item.getLocation();
+        return location != null && locationFunction.apply(location).equals(servicePointId);
```

Comment:
`locationFunction` can return `locationId`, `libraryId`, `campusId` or `institutionId`. Why do we compare them all to `servicePointId`?

Reply:
due to Description section - When more than one item is available to fulfill a page request, FOLIO should choose the item from the requesters pick up location if possible.
If there is no matching item in requested pick up location, then FOLIO should follow Library, Campus and then Institution

Reply:
codition `locationFunction.apply(location).equals(servicePointId)` will never be true. Service Point is a separate entity, it will never have ID of location, library, campus or institution.